### PR TITLE
Add -Xjit:EnableCodeCacheDisclaimingSupport option

### PR DESCRIPTION
**Pull request by gita-omr**
-  add option that enables all experimental options that can help
  code cache disclaiming

### CODE REVIEW

**Comment by gita-omr** [comment]
fyi: @mpirvu

File: compiler/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -595,10 +595,15 @@ void OMR::CodeGenerator::lowerTrees()
...
+   TR::Compilation *comp = self()->comp();
+
+   bool moveMethodToColdCode = comp->getOption(TR_MoveColdMethodsToColdCode) &&
+                               (comp->getOptLevel() == cold || comp->getOptLevel() == reducedWarm);
+
+   if ((comp->getOption(TR_SplitWarmAndColdBlocks) || moveMethodToColdCode) &&
+       !comp->compileRelocatableCode())
       {
-      self()->prepareLastWarmBlockForCodeSplitting();
+      self()->prepareLastWarmBlockForCodeSplitting(moveMethodToColdCode);
```
Comment (mpirvu) [inline compiler/codegen/OMRCodeGenerator.cpp:606]: Why don't we move the AOT code entirely to the cold area?

File: compiler/control/OMROptions.cpp [hunk 2]
```diff
@@ -1003,6 +1004,7 @@ TR::OptionTable OMR::Options::_jitOptions[] = {
         TR::Options::setStaticNumeric, (intptr_t)&OMR::Options::_minProfiledCheckcastFrequency, 0, "F%d", NOT_IN_SUBSET},
    {"minSleepTimeMsForCompThrottling=", "M<nnn>\tLower bound for sleep time during compilation throttling (ms)",
                                          TR::Options::setStaticNumeric, (intptr_t)&OMR::Options::_minSleepTimeMsForCompThrottling, 0, "F%d", NOT_IN_SUBSET },
+   {"moveColdMethodsToColdCode", "M\tmove out-of-line instructions to after last warm instruction", SET_OPTION_BIT(TR_MoveColdMethodsToColdCode), "F"},
```
Comment (mpirvu) [inline compiler/control/OMROptions.cpp:1007]: The text describing the option is not correct.

File: compiler/control/OMROptions.cpp [hunk 3]
```diff
@@ -1949,6 +1951,15 @@ OMR::Options::Options(
    if (_suppressLogFileBecauseDebugObjectNotCreated && !optimizationPlan->isLogCompilation())
       _logFile = NULL;
 
+   if (self()->getOption(TR_EnableCodeCacheDisclaimingSupport))
+      {
```
Comment (mpirvu) [inline compiler/control/OMROptions.cpp:1955]: Typically we setup options internally in `jitLatePostProcess`.

File: compiler/codegen/OMRCodeGenerator.cpp (hunk 1 above)
Comment (mpirvu) [reply compiler/codegen/OMRCodeGenerator.cpp:606]: Also, if we move the entire method to the cold area, do we still need to prepareLastWarmBlockforCodeSplitting?

**Comment by gita-omr** [comment]
Addressed comments above.

File: compiler/control/OMROptions.cpp (hunk 2 above)
Comment (gita-omr) [reply compiler/control/OMROptions.cpp]: Fixed

File: compiler/control/OMROptions.cpp (hunk 3 above)
Comment (gita-omr) [reply compiler/control/OMROptions.cpp]: Fixed

File: compiler/codegen/OMRCodeGenerator.cpp (hunk 1 above)
Comment (gita-omr) [reply compiler/codegen/OMRCodeGenerator.cpp:606]: I would suggest to first see if it helps footprint and then improve if it does.

**Comment by gita-omr** [comment]
Made WIP to allocate a whole method in cold cache.

File: compiler/codegen/OMRCodeGenerator.cpp (hunk 1 above)
Comment (mpirvu) [reply compiler/codegen/OMRCodeGenerator.cpp:606]: What was the intent behind `comp->getOptLevel() == reducedWarm` ? This is an opt level that we do not use currently. There is the notion another notion of reduced-warm, but that applies to warm compilation without optServer. E.g.: 
`bool isColdOrReducedWarm = (comp->getMethodHotness() < warm) || (comp->getMethodHotness() == warm && comp->getOption(TR_NoOptServer));`

**Comment by gita-omr** [comment]
Decided to use that PR only for the -Xjit:EnableCodeCacheDisclaimingSupport option

**Review by mpirvu — APPROVED** [review]
LGTM

File: compiler/codegen/OMRCodeGenerator.cpp (hunk 1 above)
Comment (gita-omr) [reply compiler/codegen/OMRCodeGenerator.cpp]: Good to know. Thanks.

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**